index a17024462..d3ee6055e 100644
--- a/templates/skeleton/vite.config.ts
+++ b/templates/skeleton/vite.config.ts
@@ -23,7 +23,7 @@ export default defineConfig({
        * Include 'example-dep' in the array below.
        * @see https://vitejs.dev/config/dep-optimization-options
        */
-      include: ['set-cookie-parser', 'cookie', 'react-router'],
+      include: ['set-cookie-parser', 'cookie', 'react-router', 'crypto-js'],
     },
   },
   server: {
