Index: /@cloudflare/vite-plugin/dist/index.js
===================================================================
--- /@cloudflare/vite-plugin/dist/index.js
+++ /@cloudflare/vite-plugin/dist/index.js
@@ -14153,12 +14153,18 @@
     root,
     /* prefixes */
     ""
   );
-  const entryWorkerConfigPath = getValidatedWranglerConfigPath(
-    root,
-    pluginConfig.configPath
-  );
+  let entryWorkerConfigPath
+  if (pluginConfig.config != null) {
+    entryWorkerConfigPath = path8.join(root, 'wrangler-entry-worker-config-generated.json');
+    fs4.writeFileSync(entryWorkerConfigPath, JSON.stringify(pluginConfig.config, null, 2));
+  } else {
+    entryWorkerConfigPath = getValidatedWranglerConfigPath(
+      root,
+      pluginConfig.configPath
+    );
+  }
   const entryWorkerResolvedConfig = getWorkerConfig(
     entryWorkerConfigPath,
     cloudflareEnv,
     pluginConfig.experimental?.mixedMode ?? false,
