diff --git a/node_modules/@remix-run/dev/dist/vite/vite-node.js b/node_modules/@remix-run/dev/dist/vite/vite-node.js
index 7c69d5f..3885b87 100644
--- a/node_modules/@remix-run/dev/dist/vite/vite-node.js
+++ b/node_modules/@remix-run/dev/dist/vite/vite-node.js
@@ -23,7 +23,8 @@ async function createContext(viteConfig = {}) {
   let devServer = await vite.createServer(vite.mergeConfig({
     server: {
       preTransformRequests: false,
-      hmr: false
+      hmr: false,
+      ...viteConfig.server
     },
     optimizeDeps: {
       noDiscovery: true
