index b7eb81513..7a64b33e8 100644
--- a/templates/skeleton/app/root.tsx
+++ b/templates/skeleton/app/root.tsx
@@ -82,6 +82,7 @@ export async function loader(args: Route.LoaderArgs) {
   return {
     ...deferredData,
     ...criticalData,
+    selectedLocale: args.context.storefront.i18n,
     publicStoreDomain: env.PUBLIC_STORE_DOMAIN,
     publicStorefrontToken: env.PUBLIC_STOREFRONT_API_TOKEN,
     storefrontApiVersion: storefront.apiVersion,
@@ -193,7 +194,10 @@ export default function App() {
         shop={data.shop}
         consent={data.consent}
       >
-        <PageLayout {...data}>
+        <PageLayout
+          key={`${data.selectedLocale.language}-${data.selectedLocale.country}`}
+          {...data}
+        >
           <Outlet />
         </PageLayout>
       </Analytics.Provider>
