index 04188ea73..01a872b08 100644
--- a/templates/skeleton/app/routes/sitemap.$type.$page[.xml].tsx
+++ b/templates/skeleton/app/routes/sitemap.$type.$page[.xml].tsx
@@ -1,5 +1,6 @@
 import type {Route} from './+types/sitemap.$type.$page[.xml]';
 import {getSitemap} from '@shopify/hydrogen';
+import {SUPPORTED_LOCALES} from '../lib/i18n';
 
 export async function loader({
   request,
@@ -10,7 +11,9 @@ export async function loader({
     storefront,
     request,
     params,
-    locales: ['EN-US', 'EN-CA', 'FR-CA'],
+    locales: SUPPORTED_LOCALES.map(
+      (locale) => `${locale.language}-${locale.country}`,
+    ),
     getLink: ({type, baseUrl, handle, locale}) => {
       if (!locale) return `${baseUrl}/${type}/${handle}`;
       return `${baseUrl}/${locale}/${type}/${handle}`;
