index d7877d604..4c1f09986 100644
--- a/templates/skeleton/app/lib/variants.ts
+++ b/templates/skeleton/app/lib/variants.ts
@@ -6,35 +6,25 @@ export function useVariantUrl(
   handle: string,
   selectedOptions?: SelectedOption[],
 ) {
-  const {pathname} = useLocation();
-
   return useMemo(() => {
     return getVariantUrl({
       handle,
-      pathname,
       searchParams: new URLSearchParams(),
       selectedOptions,
     });
-  }, [handle, selectedOptions, pathname]);
+  }, [handle, selectedOptions]);
 }
 
 export function getVariantUrl({
   handle,
-  pathname,
   searchParams,
   selectedOptions,
 }: {
   handle: string;
-  pathname: string;
   searchParams: URLSearchParams;
   selectedOptions?: SelectedOption[];
 }) {
-  const match = /(\/[a-zA-Z]{2}-[a-zA-Z]{2}\/)/g.exec(pathname);
-  const isLocalePathname = match && match.length > 0;
-
-  const path = isLocalePathname
-    ? `${match![0]}products/${handle}`
-    : `/products/${handle}`;
+  const path = `/products/${handle}`;
 
   selectedOptions?.forEach((option) => {
     searchParams.set(option.name, option.value);
