index b7eb81513..c265a7e69 100644
--- a/templates/skeleton/app/root.tsx
+++ b/templates/skeleton/app/root.tsx
@@ -98,6 +98,8 @@ export async function loader(args: Route.LoaderArgs) {
       country: args.context.storefront.i18n.country,
       language: args.context.storefront.i18n.language,
     },
+    // @description Add public store subdomain for metaobjects
+    publicStoreSubdomain: args.context.env.PUBLIC_STORE_DOMAIN,
   };
 }
 
