index 375fc7b04..586ac94a9 100644
--- a/templates/skeleton/app/root.tsx
+++ b/templates/skeleton/app/root.tsx
@@ -90,6 +90,8 @@ export async function loader(args: Route.LoaderArgs) {
       country: args.context.storefront.i18n.country,
       language: args.context.storefront.i18n.language,
     },
+    // @description Add public store subdomain for metaobjects
+    publictoreSubdomain: args.context.env.PUBLIC_STORE_DOMAIN,
   };
 }
 
