index 70287741..3a2f0b24 100644
--- a/templates/skeleton/app/components/CartMain.tsx
+++ b/templates/skeleton/app/components/CartMain.tsx
@@ -1,14 +1,15 @@
-import {useOptimisticCart} from '@shopify/hydrogen';
+import {useOptimisticCart, type OptimisticCart} from '@shopify/hydrogen';
 import {Link} from '@remix-run/react';
 import type {CartApiQueryFragment} from 'storefrontapi.generated';
 import {useAside} from '~/components/Aside';
 import {CartLineItem} from '~/components/CartLineItem';
 import {CartSummary} from './CartSummary';
+import {DeliveryAddressManager} from './DeliveryAddressManager';
 
 export type CartLayout = 'page' | 'aside';
 
 export type CartMainProps = {
-  cart: CartApiQueryFragment | null;
+  cart: CartApiQueryFragment | OptimisticCart | null;
   layout: CartLayout;
 };
 
@@ -26,7 +27,7 @@ export function CartMain({layout, cart: originalCart}: CartMainProps) {
     cart &&
     Boolean(cart?.discountCodes?.filter((code) => code.applicable)?.length);
   const className = `cart-main ${withDiscount ? 'with-discount' : ''}`;
-  const cartHasItems = cart?.totalQuantity ? cart.totalQuantity > 0 : false;
+  const cartHasItems = cart?.totalQuantity && cart?.totalQuantity > 0;
 
   return (
     <div className={className}>
@@ -39,7 +40,12 @@ export function CartMain({layout, cart: originalCart}: CartMainProps) {
             ))}
           </ul>
         </div>
-        {cartHasItems && <CartSummary cart={cart} layout={layout} />}
+        {cartHasItems && (
+          <>
+            <CartSummary cart={cart} layout={layout} />
+            <DeliveryAddressManager cart={cart as OptimisticCart} />
+          </>
+        )}
       </div>
     </div>
   );
