index dc4426a9..84666e74 100644
--- a/templates/skeleton/app/lib/fragments.ts
+++ b/templates/skeleton/app/lib/fragments.ts
@@ -47,6 +47,10 @@ export const CART_QUERY_FRAGMENT = `#graphql
           title
           id
           vendor
+          options {
+            name
+            values
+          }
         }
         selectedOptions {
           name
@@ -55,54 +59,68 @@ export const CART_QUERY_FRAGMENT = `#graphql
       }
     }
   }
-  fragment CartLineComponent on ComponentizableCartLine {
+  fragment CartAddress on CartSelectableAddress {
     id
-    quantity
-    attributes {
-      key
-      value
-    }
-    cost {
-      totalAmount {
-        ...Money
-      }
-      amountPerQuantity {
-        ...Money
-      }
-      compareAtAmountPerQuantity {
-        ...Money
+    oneTimeUse
+    selected
+    address {
+      ...on CartDeliveryAddress {
+        formatted
+        formattedArea
+        address1
+        address2
+        city
+        countryCode
+        company
+        firstName
+        lastName
+        name
+        phone
+        provinceCode
+        zip
       }
     }
-    merchandise {
-      ... on ProductVariant {
-        id
-        availableForSale
-        compareAtPrice {
-          ...Money
-        }
-        price {
-          ...Money
-        }
-        requiresShipping
-        title
-        image {
-          id
-          url
-          altText
-          width
-          height
-        }
-        product {
-          handle
-          title
-          id
-          vendor
-        }
-        selectedOptions {
-          name
-          value
-        }
+  }
+  fragment DeliveryGroup on CartDeliveryGroup {
+    id
+    deliveryAddress {
+      formatted
+      formattedArea
+      address1
+      address2
+      city
+      country
+      countryCode: countryCodeV2
+      company
+      firstName
+      id
+      lastName
+      name
+      phone
+      province
+      provinceCode
+      zip
+    }
+    deliveryOptions {
+      code
+      deliveryMethodType
+      description
+      estimatedCost {
+        amount
       }
+      handle
+      title
+    }
+    groupType
+    selectedDeliveryOption {
+      code
+      deliveryMethodType
+      description
+      estimatedCost {
+        amount
+      }
+      handle
+      title
     }
   }
   fragment CartApiQuery on Cart {
@@ -132,9 +150,6 @@ export const CART_QUERY_FRAGMENT = `#graphql
       nodes {
         ...CartLine
       }
-      nodes {
-        ...CartLineComponent
-      }
     }
     cost {
       subtotalAmount {
@@ -159,6 +174,16 @@ export const CART_QUERY_FRAGMENT = `#graphql
       code
       applicable
     }
+    delivery {
+      addresses {
+        ...CartAddress
+      }
+    }
+    deliveryGroups(first: 10) {
+      nodes {
+        ...DeliveryGroup
+      }
+    }
   }
 ` as const;
 
