index 29e641f48..57ade2a02 100644
--- a/templates/skeleton/app/routes/account.tsx
+++ b/templates/skeleton/app/routes/account.tsx
@@ -1,6 +1,8 @@
-
-import {data as remixData, type LoaderFunctionArgs} from '@shopify/remix-oxygen';
-import { Form, NavLink, Outlet, useLoaderData } from 'react-router';
+import {
+  data as remixData,
+  type LoaderFunctionArgs,
+} from '@shopify/remix-oxygen';
+import {Form, NavLink, Outlet, useLoaderData} from 'react-router';
 import {CUSTOMER_DETAILS_QUERY} from '~/graphql/customer-account/CustomerDetailsQuery';
 
 export function shouldRevalidate() {
@@ -72,6 +74,10 @@ function AccountMenu() {
         Orders &nbsp;
       </NavLink>
       &nbsp;|&nbsp;
+      <NavLink to="/account/subscriptions" style={isActiveStyle}>
+        &nbsp; Subscriptions &nbsp;
+      </NavLink>
+      &nbsp;|&nbsp;
       <NavLink to="/account/profile" style={isActiveStyle}>
         &nbsp; Profile &nbsp;
       </NavLink>
