index f6b6b2a3..f3f9ecff 100644
--- a/templates/skeleton/app/lib/fragments.ts
+++ b/templates/skeleton/app/lib/fragments.ts
@@ -59,6 +59,11 @@ export const CART_QUERY_FRAGMENT = `#graphql
         id
       }
     }
+    sellingPlanAllocation {
+      sellingPlan {
+         name
+      }
+    }
   }
   fragment CartLineComponent on ComponentizableCartLine {
     id
@@ -112,6 +117,11 @@ export const CART_QUERY_FRAGMENT = `#graphql
     lineComponents {
       ...CartLine
     }
+    sellingPlanAllocation {
+      sellingPlan {
+         name
+      }
+    }
   }
   fragment CartApiQuery on Cart {
     updatedAt
