index 1ebce62d3..3a7387e89 100644
--- a/templates/skeleton/vite.config.ts
+++ b/templates/skeleton/vite.config.ts
@@ -1,19 +1,20 @@
 import {defineConfig} from 'vite';
 import {hydrogen} from '@shopify/hydrogen/vite';
-import {oxygen} from '@shopify/mini-oxygen/vite';
 import {reactRouter} from '@react-router/dev/vite';
 import tsconfigPaths from 'vite-tsconfig-paths';
 
 export default defineConfig({
-  plugins: [hydrogen(), oxygen(), reactRouter(), tsconfigPaths()],
+  plugins: [
+    hydrogen(),
+    reactRouter(),
+    tsconfigPaths(),
+  ],
   build: {
     // Allow a strict Content-Security-Policy
     // withtout inlining assets as base64:
     assetsInlineLimit: 0,
+    target: 'esnext',
   },
-
-  // Default React Router 7.8.x structure (no custom environment config)
-
   ssr: {
     optimizeDeps: {
       /**
@@ -26,10 +27,7 @@ export default defineConfig({
        * Include 'example-dep' in the array below.
        * @see https://vitejs.dev/config/dep-optimization-options
        */
-      include: ['set-cookie-parser', 'cookie', 'react-router'],
+      include: [],
     },
   },
-  server: {
-    allowedHosts: ['.tryhydrogen.dev'],
-  },
 });
