index c9538bf4..69ce3030 100644
--- a/templates/skeleton/env.d.ts
+++ b/templates/skeleton/env.d.ts
@@ -19,7 +19,8 @@ declare global {
   const process: {env: {NODE_ENV: 'production' | 'development'}};
 
   interface Env extends HydrogenEnv {
-    // declare additional Env parameter use in the fetch handler and Remix loader context here
+    // declare additional Env parameter in the fetch handler and in Remix loader context here
+    GTM_CONTAINER_ID: `GTM-${string}`;
   }
 }
 
